### DISCUSSION TITLE
Seeking Guidance on Connecting and Controlling Servo Motor via Crazyflie Board and Quad Decks

### DISCUSSION THREAD

**Discussion by NarekHarutyu** [bitcraze/discussions#780] [Q&A]
Hello, I am currently working on connecting a servo motor to the Crazyflie board, which is connected to both a bigger quad deck and a prototyping deck. I have successfully connected the power and ground wires of the servo motor to the corresponding holes on the prototyping deck, which are linked to the ground and VCOM of the Crazyflie board. However, I am unsure about the appropriate connection point for the signal wire of the servo motor in order to control it from the Crazyflie board.
Could you kindly provide guidance on which specific pin I should use on either the Crazyflie board, prototyping deck, or big quad deck to effectively send signals from the Crazyflie board and control the servo movement? Please note that the Crazyflie board and big quad deck are utilized for flying a larger quadcopter.
Thank you for your assistance.

**Comment by simondlevy** [comment]
I'm not part of the Crazyflie development team, but looking at the pinout on page 2 of the Crazyflie2.1 [datasheet](https://www.bitcraze.io/documentation/hardware/crazyflie_2_1/crazyflie_2_1-datasheet.pdf), I believe that you'd want to use one of the three PWM pins in the upper-right corner of the board.  I'm assuming that your servo is a traditional PWM servo; i.e., it's not using the SBUS protocol.

**Reply by NarekHarutyu** [reply]
Thank you very much for your response. It seems very reasonable to connect the signal wire of the servo to the TX pin. Do you know, however, how to send PWM signals to the TX pin from the firmware?

**Reply by simondlevy** [reply]
Unfortunately my microcontroller experience has mostly been with the Arduino core, in which operations like this take only one or two lines of code.  This [tutorial](https://stm32f4-discovery.net/2014/05/stm32f4-stm32f429-discovery-pwm-tutorial/) looks like a good place to learn how to do it using the STM32F4 HAL (Hardware Abstraction Layer).  Looking at the Crazyflie [firmware](https://github.com/bitcraze/crazyflie-firmware/blob/41a9c97569e565e927a0fe3b8f032dbcdb03aa1b/src/drivers/src/motors.c) for running the motors might also provide some direction.
